Library/Client.ts

```diff
@@ -43,7 +43,7 @@ class Client {
         this.context = new Context();
         this.commonProperties = {};
 
-        var sender = new Sender(() => config.endpointUrl);
+        var sender = new Sender(this.config);
```

Comment:
Are there side effects to removing the input function? I imagine the inputs to this and channel were all wrapped inside functions for some reason

Reply:
I think they were wrapped, so that sender always has access to the the latest (possibly updated) endpointUrl and doesn't cache it as local variable. This delayed retrieval is still preserved as we are only getting url from inside `send()` method.

Reply:
Couldn't ikey still change however?